codelists/views.py

```diff
@@ -26,30 +38,87 @@ def index(request):
     return render(request, "codelists/index.html", ctx)
 
 
-@login_required
-def create_codelist(request, project_slug):
-    project = get_object_or_404(Project, slug=project_slug)
-    if request.method == "POST":
-        form = CodelistForm(request.POST, request.FILES)
-        if form.is_valid():
-            codelist = actions.create_codelist(
-                project=project,
-                name=form.cleaned_data["name"],
-                coding_system_id=form.cleaned_data["coding_system_id"],
-                description=form.cleaned_data["description"],
-                methodology=form.cleaned_data["methodology"],
-                csv_data=form.cleaned_data["csv_data"].read().decode("utf8"),
-            )
-            return redirect(codelist)
-    else:
-        form = CodelistForm()
+@method_decorator(login_required, name="dispatch")
+class CreateCodelist(TemplateView):
```

Comment:
I'm not used to CBVs, so I don't find this particularly easy to read.  One change that might help would be to reorder the methods like this:

* dispatch
* post
* all_valid/some_valid
* get_context_data
* get_form, get_X_formset

You can then more-or-less read the code top to bottom.

Reply:
Done in 2aff811